File: Maths/matrices.js
/**
 * Différentes fonctions pour manipuler des matrices.
 * @author Noam Zeitoun
 * @version 1.0
 */

/** *
 * @description Matrice identité.
 * @param {number} n - Dimension de la matrice.
 * @return {Array} La matrice identité.
 */
const matriceIdentite = (n) => {
    let matrice = [];
    for (let i = 0; i < n; i++) {
        matrice[i] = [];
        for (let j = 0; j < n; j++) {
            matrice[i][j] = (i == j) ? 1 : 0;
        }
    }
    return matrice;
};

/**
 * @description Vérifie la dimension d'une matrice.
 * @param {Array} matrice - La matrice à vérifier.
 * @param {Number} n - La dimension attendue.
 * @return {boolean} - True si la matrice est valide, false sinon.
 */
const dimensionMatrice = (matrice, n) => (matrice.length == n && matrice[0].length == n);

/* ---------------- *
 * Travail en cours *
 * ---------------- */

// Tests
console.table({
    'matriceIdentite(3)': matriceIdentite(3),
    "dimensionMatrice([[0, 0, 0], [0, 0, 0], [0, 0, 0]], 3)": dimensionMatrice([[0, 0, 0], [0, 0, 0], [0, 0, 0]], 3),
})

File: Python/Maths/matrices.py
"""
Description:
    Différentes fonctions pour manipuler des matrices.

Auteur:
    Noam Zeitoun

Version:
    1.0
"""

def matrice_identite(n: int) -> list:
    """
    Description:
        Crée une matrice identité de taille n.
    
    Paramètres:
        n: {int} -- Taille de la matrice.

    Retourne:
        {list} -- Matrice identité de taille n.

    Exemple:
        >>> matrice_identite(3)
        [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
    """
    matrice = []
    for i in range(n):
        matrice.append([])
        for j in range(n):
            if i == j:
                matrice[i].append(1)
            else:
                matrice[i].append(0)
    return matrice

def dimension_matrice(matrice: list, n: int) -> bool:
    """
    Description:
        Vérifie si la matrice est de taille n.
    
    Paramètres:
        matrice: {list} -- Matrice à vérifier.
        n: {int} -- Taille de la matrice.

    Retourne:
        {bool} -- True si la matrice est de taille n, False sinon.
            
    Exemple:
        >>> dimension_matrice([[0, 0, 0], [0, 0, 0], [0, 0, 0]], 3)
        True
    """
    if len(matrice) == n and len(matrice[0]) == n:
        return True
    else:
        return False

# TESTS
print(
    'matrice_identite(3)'+' ----------------------------> ', matrice_identite(3),
    '\ndimension_matrice(matrice_identite(3), 3)'+' ------> ', dimension_matrice(matrice_identite(3), 3)
)